src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/Microsoft.Diagnostics.Monitoring.TestCommon.csproj

```diff
@@ -1,7 +1,51 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
   </PropertyGroup>
 
+  <!-- 
+    DARCVersionFileName, GenerateDARCVersionCs, and InjectDARCVersion are used for injecting the value of the Microsoft.NETCore.App version
+    into csharp code so that it can be used by testing harnesses for running tests under a specific runtime version.  This specifically
+    allows tests to run on a version of the runtime besides whatever was included with Arcade's SDK.
+   -->
+
+  <PropertyGroup>
+    <DARCVersionFileName>$(IntermediateOutputPath)/$(TargetFramework)/DARCVersion.g.cs</DARCVersionFileName>
+    <DARCRuntimeVersion Condition="'$(TargetFramework)' == 'netcoreapp3.1'">$(MicrosoftNETCoreApp31Version)</DARCRuntimeVersion>
+    <DARCRuntimeVersion Condition="'$(TargetFramework)' == 'net5.0'">$(MicrosoftNETCoreApp50Version)</DARCRuntimeVersion>
+  </PropertyGroup>
+
+  <Target Name="GenerateDARCVersionCs" Inputs="$(VersionsPropsPath)" Outputs="$(DARCVersionFileName)">
+    <PropertyGroup>
+      <DARCVersionFile>
+        <![CDATA[
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+// THIS FILE IS AUTO-GENERATED DURING BUILD.  MODIFICATIONS _WILL_ BE OVERWRITTEN!
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon
```

Comment:
nit: can we put the contents of this header outside the csproj file, or at least in a separate targets file?

Reply:
I could create a template file for this.

Reply:
I might look into creating a source generator: https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md